internal/services/compute/virtual_machine.go

```diff
@@ -258,9 +270,14 @@ func expandVirtualMachineOSDisk(input []interface{}, osType compute.OperatingSys
 	}
 
 	if diffDiskSettingsRaw := raw["diff_disk_settings"].([]interface{}); len(diffDiskSettingsRaw) > 0 {
+		if caching != string(compute.CachingTypesReadOnly) {
```

Comment:
Can you please add a link above pointing to the document about this restriction?